### PR TITLE
[SPARK-43762][SPARK-43763][SPARK-43764][SPARK-43765][SPARK-43766][SQL] Assign names to the error class _LEGACY_ERROR_TEMP_24[06-10]

### PR DESCRIPTION
**Pull request by beliefer**
### What changes were proposed in this pull request?
The pr aims to assign a name to the error class _LEGACY_ERROR_TEMP_24[06-10].

### Why are the changes needed?
Improve the error framework.

### Does this PR introduce _any_ user-facing change?
'No'.

### How was this patch tested?
Exists test cases.

### CODE REVIEW

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala [hunk 1]
```diff
@@ -1013,13 +1013,13 @@ class AnalysisSuite extends AnalysisTest with Matchers {
 
   test("SPARK-31975: Throw user facing error when use WindowFunction directly") {
     assertAnalysisError(testRelation2.select(RowNumber()),
-      Seq("Window function row_number() requires an OVER clause."))
+      Seq("Window function \"row_number()\" requires an OVER clause."))
```
Comment (MaxGekk) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala:1016]: Please, migrate this onto `assertAnalysisErrorClass`

File: core/src/main/resources/error/error-classes.json [hunk 2]
```diff
@@ -5680,4 +5675,4 @@
       "Failed to get block <blockId>, which is not a shuffle block"
     ]
   }
-}
\ No newline at end of file
+}
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json]: Revert this back, see https://github.com/apache/spark/pull/41256

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala (hunk 1 above)
Comment (beliefer) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala:1016]: Got it.

File: core/src/main/resources/error/error-classes.json (hunk 2 above)
Comment (beliefer) [reply core/src/main/resources/error/error-classes.json]: https://github.com/apache/spark/commit/32d42bbe98da9a7e8c38b9c3187c75dbbaaaafbb removed the last empty line.
This PR restores it.

File: core/src/main/resources/error/error-classes.json [hunk 3]
```diff
@@ -2272,6 +2282,16 @@
     ],
     "sqlState" : "42P01"
   },
+  "WINDOW_FUNCTION_AND_FRAME_MISMATCH" : {
+    "message" : [
+      "<wf> function can only be evaluated in an ordered row-based window frame with a single offset: <w>."
```
Comment (MaxGekk) [inline core/src/main/resources/error/error-classes.json:2287]: Let's assign proper names of parameters here and other places.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala [hunk 4]
```diff
@@ -1183,7 +1192,7 @@ class AnalysisSuite extends AnalysisTest with Matchers {
         |    GROUP BY a, b
         |)
       """.stripMargin),
-      Seq("grouping_id() can only be used with GroupingSets/Cube/Rollup"),
+      Seq("grouping()/grouping_id() can only be used with GroupingSets/Cube/Rollup"),
```
Comment (MaxGekk) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala:1195]: Use the function which checks the error class and its parameters.

**Review by MaxGekk — CHANGES_REQUESTED** [review]

File: core/src/main/resources/error/error-classes.json [hunk 5]
```diff
@@ -5757,4 +5752,4 @@
       "Failed to get block <blockId>, which is not a shuffle block"
     ]
   }
-}
\ No newline at end of file
+}
```
[the remark this answers is not in the document]
Comment (beliefer) [reply core/src/main/resources/error/error-classes.json:5755]: I don't know the background.
Please see https://github.com/apache/spark/pull/41279#discussion_r1205260168.

File: core/src/main/resources/error/error-classes.json (hunk 3 above)
Comment (beliefer) [reply core/src/main/resources/error/error-classes.json:2287]: How about `WINDOW_SPEC_MISSING_ORDER_BY_OR_WITHOUT_SINGLE_OFFSET` ?
or `INVALID_WINDOW_SPEC_FOR_FRAME_LESS_OFFSET_WINDOW_FUNCTION`.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala (hunk 4 above)
Comment (beliefer) [reply sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisSuite.scala:1195]: Got it.

File: core/src/main/resources/error/error-classes.json [hunk 6]
```diff
@@ -175,6 +175,11 @@
     ],
     "sqlState" : "22018"
   },
+  "CAST_IS_UNRESOLVED" : {
```
[the remark this answers is not in the document]
Comment (beliefer) [reply core/src/main/resources/error/error-classes.json:178]: Got it.

File: core/src/main/resources/error/error-classes.json [hunk 7]
```diff
@@ -573,6 +578,11 @@
     ],
     "sqlState" : "42704"
   },
+  "DISTINCT_WINDOW_FUNCTION_UNSUPPORTED" : {
```
[the remark this answers is not in the document]
Comment (beliefer) [reply core/src/main/resources/error/error-classes.json:581]: User can see.

File: sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala [hunk 8]
```diff
@@ -193,7 +193,12 @@ class AnalysisErrorSuite extends AnalysisTest {
           UnresolvedAttribute("a") :: Nil,
           SortOrder(UnresolvedAttribute("b"), Ascending) :: Nil,
           UnspecifiedFrame)).as("window")),
-    "Distinct window functions are not supported" :: Nil)
+    errorClass = "DISTINCT_WINDOW_FUNCTION_UNSUPPORTED",
```
Comment (beliefer) [inline sql/catalyst/src/test/scala/org/apache/spark/sql/catalyst/analysis/AnalysisErrorSuite.scala:196]: @MaxGekk There has a test for `DISTINCT_WINDOW_FUNCTION_UNSUPPORTED`. Do we really convert it as internal error ?

File: core/src/main/resources/error/error-classes.json (hunk 7 above)
[the remark this answers is not in the document]
Comment (MaxGekk) [reply core/src/main/resources/error/error-classes.json:576]: ahh, ok. Let's leave it as is.

**Review by MaxGekk — APPROVED** [review]

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master.
Thank you, @beliefer.

**Comment by beliefer** [comment]
@MaxGekk Thank you!

**Pull request closed without merging**